lib/holidays.rb

```diff
@@ -160,6 +160,50 @@ def next_holidays(holidays_count, options, from_date = Date.today)
       UseCaseFactory.next_holiday.call(holidays_count, from_date, date_driver_hash, regions, observed, informal)
     end
 
+    # Get all holidays occuring from date to end of year, inclusively. 
+    #
+    # Returns an array of hashes or nil. 
+    #
+    # Incoming arguments are below: 
+    # [<tt>options</tt>]  One or more region symbols, <tt>:informal</tt> and/or <tt>:observed</tt>.
+    # [<tt>from_date</tt>]    Ruby Date object. This is an optional param, defaulted today.
+    #
+    # ==== Example
+    #   Date.today
+    #   => Tue, 23 Feb 2016
+    # 
+    #   regions = [:ca_on]
+    # 
+    #   Holidays.year_holidays(regions)
+    #   => [{:name=>"Good Friday",...},
+    #       {name=>"Easter Sunday",...},
+    #       {:name=>"Victoria Day",...},
+    #       {:name=>"Canada Day",...},
+    #       {:name=>"Civic Holiday",...},
+    #       {:name=>"Labour Day",...},
+    #       {:name=>"Thanksgiving",...},
+    #       {:name=>"Remembrance Day",...},
+    #       {:name=>"Christmas Day",...},
+    #       {:name=>"Boxing Day",...}]
+    def year_holidays(options, from_date = Date.today)
+      raise ArgumentError if options.empty?
+      raise ArgumentError unless options.is_a?(Array)
+
+      # remove the timezone
+      from_date = from_date.new_offset(0) + from_date.offset if from_date.respond_to?(:new_offset)
+
+      from_date = get_date(from_date)
+      to_date = from_date.to_datetime.change(:month => 12).end_of_month.to_date
```

Comment:
I like this but does this seems to require that a user add the core extension for `Date`. That's a change since we don't require the core extension otherwise as far as I know.

This is what I got when I tried to run locally:

```sh
% be rake console
irb -rubygems -I lib -r holidays.rb
irb(main):001:0> Holidays.year_holidays([:us])
NoMethodError: undefined method `change' for #<DateTime:0x00561d9fd08528>
        from /home/vagrant/dev/holidays/lib/holidays.rb:196:in `year_holidays'
        from (irb):1
        from /home/vagrant/.rbenv/versions/2.2.2/bin/irb:11:in `<main>'
irb(main):002:0>
```

This is a problem. Is there a way that we can get the last day of the year without requiring the core extension? Perhaps:

```ruby
Date.new(y=Date.today.year,m=12,d=31).yday
```

Or something along those lines?